packages/core-data/src/private-selectors.ts

```diff
@@ -105,3 +106,129 @@ export function getEntityRecordPermissions(
 export function getRegisteredPostMeta( state: State, postType: string ) {
 	return state.registeredPostMeta?.[ postType ] ?? {};
 }
+
+interface SiteData {
+	show_on_front?: string;
+	page_on_front?: string | number;
+	page_for_posts?: string | number;
+}
+
+export const getHomePage = createRegistrySelector( ( select ) => () => {
+	const siteData = select( STORE_NAME ).getEntityRecord( 'root', 'site' ) as
+		| SiteData
+		| undefined;
+	const homepageId =
+		siteData?.show_on_front === 'page' &&
+		[ 'number', 'string' ].includes( typeof siteData.page_on_front ) &&
+		siteData.page_on_front &&
+		!! +siteData.page_on_front // We also need to check if it's not zero(`0`).
+			? siteData.page_on_front.toString()
+			: null;
+	const isSiteLoaded = !! siteData;
+	if ( isSiteLoaded && homepageId ) {
+		return { postType: 'page', postId: homepageId };
+	}
+	if ( isSiteLoaded && ! homepageId ) {
+		const frontPageTemplateId = select( STORE_NAME ).getDefaultTemplateId( {
+			slug: 'front-page',
+		} );
+		return { postType: 'wp_template', postId: frontPageTemplateId };
+	}
+	return null;
+} );
```

Comment:
While reviewing the previous PR I was playing locally with this very "compressed" code and came up with this util function that both selectors (home page/posts page) could use:
```js
function pageId( value ) {
  if ( ! [ 'number', 'string' ].includes( typeof value ) ) {
    return null;
  }

  // We also need to check if it's not zero (`'0'`).
  if ( Number( value ) === 0 ) {
    return null;
  }

  return value.toString();
}
```
Also, in this selector you can return `null` early if `! siteData` and it greatly simplifies the code that follows: no `?.` operators and extra conditions.